src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -388,43 +399,83 @@ public void onUploadFinishedEVent(final String photoId) {
         flickrPhotoClient.setProperties(oauthRequest.getParams(), new Callback<String>() {
             @Override
             public void success(String s, Response response) {
-                setPhotoTag(photoId);
+                try {
+                    setPhotoTag(photoId);
+                } catch (UnsupportedEncodingException e) {
+                }
             }
 
             @Override
             public void failure(RetrofitError error) {
+                Toast.makeText(PhotoActivity.this, "Erreur lors de l'envoi de la photo, réessayer", Toast.LENGTH_LONG).show();
                 progressDialog.dismiss();
             }
         });
     }
 
-    private void setPhotoTag(final String photoId) {
-        OAuthRequest oauthRequest = new OAuthRequest(configManager.getFlickrApiKey(), configManager.getFlickrApiKeySecret());
+    private void setPhotoTag(final String photoId) throws UnsupportedEncodingException {
+        //Create the machine tag for Flickr to associate picture to this Poi
+        Poi currentPoi = application.getOsmTemplateComponent().getPoiManager().queryForId(poiId);
+        String flickrOsmTag = "openstreetmap";
+        if (currentPoi.getBackendId() != null && currentPoi.getBackendId().length() > 0) {
+            flickrOsmTag += ",osm:" + ((currentPoi.getWay()) ? "way" : "node") + "=" + currentPoi.getBackendId() + "";
```

Comment:
use StringBuilder to avoid the definition of too many strings in memory.
(concatenation of A+B+C+D creates A, B, C, D, A+B, A+B+C, A+B+C+D)